discovery-leap-of-faith.md

```diff
@@ -0,0 +1,38 @@
+
+# Discovery using the cloud and leap-of-faith
+## NUPNP + plex + LE
+ * (less-secure option when QR codes cannot be used)
+ * con: you need to trust not only the Box and the client, but also the bridge/DNS-server
+ * pro: the user does not have to click the (confusing) "Allow" button in the getUserMedia dialog
+ * con: lack of QR-code means trusting that no attacker is on local network during pairing phase
+
+## Flow:
+1. At build time, a serial number and a deploy token is put onto the box, so that it can request two services:
+ * setting up a DNS entry,
+ * signing a CSR.
+2. Still at build time, the serial number is put into a string of the form "https://<serial-no>.knilxof-local.org/". This string is printed onto a sticker.
+3. The sticker is put onto the physical Box. The Box is then packaged in such a way that you can't see the serial number sticker in the shop without opening the packaging.
+4. The user unpacks the Box, and plugs it into their router using an ethernet cable.
+5. The Box obtains a local IP address using DHCP, and generates two CSRs.
+6. The Box connects to the Bridge on e.g. https://register.knilfox.org and posts the following data:
+ * its serial number,
+ * the deploy token,
+ * its CSR,
+ * its local IP address.
+7. The Bridge registers two LetsEncrypt certs (one for <serial-no>.knilxof-local.org and one for <serial-no>.knilxof-remote.org), telling the Bridge to update DNS with the ACME DNS challenge strings (currently in staging, see https://community.letsencrypt.org/t/dns-challenge-is-in-staging/8322 ).
+8. The Bridge adds an A record for <serial-no>.knilfox-local.org, pointing to the Box's local IP address; <serial-no>.knilxof-remote.org points to the public end of the tunnel.
+9. The user can use the normal browser on their client to open our web app.
+10. The Box posts its long URL "https://<serial-no>.knilxof-local.org/" to https://discover.knilxof.org/I-am-a-Box.
+11. The Bridge adds an entry to its database linking the outgoing IP address of the Box to its long URL.
+12. The user types https://discover.knilxof.org/I-am-a-client.html into their browser's address bar, and the Bridge does a database lookup;
```

Comment:
I believe that you should mention explicitly that it's not supported (yet).

As a side-note, physical side-channels can probably be used for attacks, too. e.g. someone stealing control to a friend's ip cam, by just taking a picture while the friend is in the bathroom.